### ISSUE TITLE
ListViewHitTestLocations need to be updated for new ComCtl values

### ISSUE THREAD

**Issue by hughbe** [dotnet/winforms#2629]
# Problem
- We're missing the following definitions from `ListViewHitTestLocations`. These should be added to the public enum because currently we may have unknown values in `ListView.HitTest`

Name | Description
----- | -----
LVHT_EX_GROUP_HEADER | Windows Vista. The point is within the group header.
LVHT_EX_GROUP_FOOTER | Windows Vista. The point is within the group footer.
LVHT_EX_GROUP_COLLAPSE | Windows Vista. The point is within the collapse/expand button of the group.
LVHT_EX_GROUP_BACKGROUND | Windows Vista. The point is within the area of the group where items are displayed.
LVHT_EX_GROUP_STATEICON | Windows Vista. The point is within the state icon of the group.
LVHT_EX_GROUP_SUBSETLINK | Windows Vista. The point is within the subset link of the group.
LVHT_EX_GROUP | Windows Vista. LVHT_EX_GROUP_BACKGROUND \| LVHT_EX_GROUP_COLLAPSE \| LVHT_EX_GROUP_FOOTER \| LVHT_EX_GROUP_HEADER \| LVHT_EX_GROUP_STATEICON \| LVHT_EX_GROUP_SUBSETLINK.
LVHT_EX_ONCONTENTS | Windows Vista. The point is within the icon or text content of the item and not on the background.
LVHT_EX_FOOTER

**Comment by RussKie** [comment]
We have a number of definitions that are missing members. And now we are able to bridge the gap that .NET Framework could not.

**Comment by elachlan** [comment]
I ran into an issue with this, since `ListViewHitTestLocations` is `INT`, it can't handle the larger value for group background.
`LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_BACKGROUND = 0x80000000`

@merriemcgaw would it be a breaking change to change it to `UINT`? When I tried I get a CS3009 analyzer warning.
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs3009
```csharp
[Flags]
public enum ListViewHitTestLocations : uint
{
    None = (int)LVHITTESTINFO_FLAGS.LVHT_NOWHERE,
    AboveClientArea = 0x0100,
    BelowClientArea = (int)LVHITTESTINFO_FLAGS.LVHT_BELOW,
    LeftOfClientArea = (int)LVHITTESTINFO_FLAGS.LVHT_TOLEFT,
    RightOfClientArea = (int)LVHITTESTINFO_FLAGS.LVHT_TORIGHT,
    Image = (int)LVHITTESTINFO_FLAGS.LVHT_ONITEMICON,
    StateImage = 0x0200,
    Label = (int)LVHITTESTINFO_FLAGS.LVHT_ONITEMLABEL,
    GroupHeader = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_HEADER,
    GroupFooter = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_FOOTER,
    GroupExpandCollapse = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_COLLAPSE,
    GroupBackground = LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_BACKGROUND,
    GroupStateImage = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_STATEICON,
    GroupSubsetLink = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_SUBSETLINK,
    Group = GroupHeader | GroupFooter | GroupExpandCollapse | GroupBackground | GroupStateImage | GroupSubsetLink,
    Contents = (int)LVHITTESTINFO_FLAGS.LVHT_EX_ONCONTENTS,
    Footer = (int)LVHITTESTINFO_FLAGS.LVHT_EX_FOOTER,
}
```

Also `ListViewHitTestLocations.StateImage` doesn't map to `LVHT_ONITEMSTATEICON` for some reason. Its `0x0200` instead of `LVHT_ONITEMSTATEICON = 0x00000008`.

**Comment by merriemcgaw** [comment]
@JeremyKuhne might have some thoughts here.

**Comment by JeremyKuhne** [comment]
We cannot change the base type of existing enums. You can use the other values, you just need to do `unchecked((int)0x80000000)`.

Additionally, we cannot add values without going through API review first. You can turn this into a formal API Proposal and we can submit it. I presume this is pretty straightforward as Vista is well below our minimum requirements.

**Comment by elachlan** [comment]
@JeremyKuhne should this issue be used as the API suggestion with my code change above adjusted based on your suggestion?

**Comment by JeremyKuhne** [comment]
Yes, it's fine to use this one, it just needs updated to match the API Proposal requirements. You can see what is required by looking at the [runtime template](https://github.com/dotnet/runtime/issues/new?assignees=&labels=api-suggestion&template=02_api_proposal.yml&title=%5BAPI+Proposal%5D%3A+).

@dreddy-work It looks like we might want to update our API template in WinForms to have the right format for API submission. They are no longer just pointing to a document.

**Comment by elachlan** [comment]
**Background and motivation**
`ListViewHitTestLocations` is currently missing some values, the values should be updated to match the values specified in [LVHITTESTINFO](https://learn.microsoft.com/windows/win32/api/commctrl/ns-commctrl-lvhittestinfo). This will allow us to handle them in `ListView.HitTest`.

These values were missing because .NET framework supported Windows version older than Vista.

**API Proposal**
```
[Flags]
public enum ListViewHitTestLocations : uint
{
    None = (int)LVHITTESTINFO_FLAGS.LVHT_NOWHERE,
    AboveClientArea = 0x0100,
    BelowClientArea = (int)LVHITTESTINFO_FLAGS.LVHT_BELOW,
    LeftOfClientArea = (int)LVHITTESTINFO_FLAGS.LVHT_TOLEFT,
    RightOfClientArea = (int)LVHITTESTINFO_FLAGS.LVHT_TORIGHT,
    Image = (int)LVHITTESTINFO_FLAGS.LVHT_ONITEMICON,
    StateImage = 0x0200,
    Label = (int)LVHITTESTINFO_FLAGS.LVHT_ONITEMLABEL,
    GroupHeader = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_HEADER,
    GroupFooter = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_FOOTER,
    GroupExpandCollapse = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_COLLAPSE,
    GroupBackground = unchecked((int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_BACKGROUND),
    GroupStateImage = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_STATEICON,
    GroupSubsetLink = (int)LVHITTESTINFO_FLAGS.LVHT_EX_GROUP_SUBSETLINK,
    Group = GroupHeader | GroupFooter | GroupExpandCollapse | GroupBackground | GroupStateImage | GroupSubsetLink,
    Contents = (int)LVHITTESTINFO_FLAGS.LVHT_EX_ONCONTENTS,
    Footer = (int)LVHITTESTINFO_FLAGS.LVHT_EX_FOOTER,
}
```
**API Usage**
`ListViewHitTestLocations.GroupBackground`

**Alternative Designs**
N/A

**Risks**
Low risk as only adding missing enum values.

**Comment by merriemcgaw** [comment]
Thanks @elachlan! I copied your comment into the original issue description to make it easier for API review board. I will sync with the team and make sure we feel this is ready to submit for them and let you know how it goes.
